### ISSUE TITLE
font-lock-mode is disabled in a buggy way

### ISSUE THREAD

**Issue by NightMachinary** [hlissner/doom-emacs#4924]
If I have font-lock on and I do `M-x find-library RET simple RET`, I have colours:
[![enter image description here][1]][1]
If I then `M-x font-lock-mode` to disable the mode, the colours disappear in the currently-visible portion of the buffer:
[![enter image description here][2]][2]
But the colours are still present everywhere else if I scroll the buffer:
[![enter image description here][3]][3]

This does not happen with `emacs -Q`, but it is present with a fresh Doom emacs.

Screenshot of the bug in a fresh Doom emacs:
![image](https://user-images.githubusercontent.com/36224762/115998466-9a0fcb00-a5fc-11eb-8fd3-ad176f971c03.png)

```
GNU Emacs 27.1 (build 1, x86_64-apple-darwin20.3.0, NS appkit-2022.30 Version 11.2.1 (Build 20D75)) of 2021-03-13
```

  [1]: https://i.stack.imgur.com/y2HwN.png
  [2]: https://i.stack.imgur.com/MoBMF.png
  [3]: https://i.stack.imgur.com/0qXZf.png

**System information:**
```
  generated  Apr 25, 2021 19:30:37
     distro  MacOS 11.2.1 Darwin 20.3.0 x86_64
      emacs  27.1 ~/.emacs.d.doom/
       doom  2.0.9 HEAD -> patch1, fork/patch1 b2aab9798 2021-04-15 02:54:19 +0430 ~/doom.d/
      shell  /bin/dash
   features  RSVG IMAGEMAGICK GLIB NOTIFY KQUEUE ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS XIM NS MODULES THREADS XWIDGETS JSON PDUMPER LCMS2 GMP
     traits  batch server-running envvar-file
```

**Comment by hlissner** [comment]
I can't reproduce this on the latest commit of vanilla Doom (8bb54a95f), in 27.1, 27.2 or 28, on NixOS or Arch.

That said, Emacs does its best to defer refontification when buffers grow beyond a certain threshold (see `SPC h v ^jit-lock-` to peruse its variables). If this is important to you, either call `M-x font-lock-update` after turning off the mode, to forcibly flush the whole buffer, or tweak jit-lock's settings to reduce the time it waits to do it automatically.

As I'm not entirely convinced this should be considered a bug (or a Doom issue), I will close this, but feel free to follow up here.

**Comment by NightMachinary** [comment]
@hlissner I don't seem to have `font-lock-update`?

```
Debugger entered--Lisp error: (void-function font-lock-update)
```

I tried `font-lock-fontify-buffer` instead, and the effect is that the uncolored parts are colored again.

Apart from this `font-lock-fontify-buffer`, when I change anything or type new text, the changes are always colored. So it seems unlikely that the problem is that the colors were cached.

---

I disabled `fast-but-imprecise-scrolling`; No effect.

---

As you have not been to reproduce this, let's keep the issue closed. It's weird, since I remember testing this on a remote Ubuntu server as well, and seeing the same issue there ...
